dartagnan/src/main/java/com/dat3m/dartagnan/program/event/common/FenceBase.java

```diff
@@ -2,20 +2,19 @@
 
 import com.dat3m.dartagnan.program.event.Tag;
 import com.dat3m.dartagnan.program.event.core.AbstractEvent;
+import com.google.common.base.Preconditions;
 
 @NoInterface
 public abstract class FenceBase extends AbstractEvent {
 
     protected final String name;
-    protected final String mo; // May be NULL or empty for fences that do not support a mo.
+    protected final String mo; // For fences that do not support a mo, use Fence instead.
```

Comment:
The comment is outdated.

Reply:
No ... these are language level fences and I argue that for those `mo` is mandatory (which is aligned with the fact we have `Preconditions.checkArgument(!mo.isEmpty());` in the constructor)

Reply:
`Fence` does not exist anymore, hence it is outdated. Strictly speaking, this class is also not just for language-level fences, it is just a common(!) implementation that can be reused for whatever purposes.
Whether all fences have a `mo`or not is likely a matter of convention. Intuitively, I would have said a compiler barrier has no memory ordering, but in our `linux.h` we do model compiler barriers as a fence with `barrier` memory ordering. I'm fine with that, though I don't necessarily see why `FenceBase` should enforce that.
After all, this class is abstract and any subclass is free to decide if it needs a memory ordering or not.